src/shame.js

```diff
@@ -43,3 +43,42 @@ function clearAxisTypes(gd, traces) {
     }
   }
 }
+
+export const shamefullyAdjustAxisRef = (graphDiv, payload) => {
+  if (payload.tracesNeedingAxisAdjustment) {
+    payload.tracesNeedingAxisAdjustment.forEach(trace => {
+      const axis = trace[payload.axisAttrToAdjust].charAt(0);
+      const currentAxisIdNumber = Number(
+        trace[payload.axisAttrToAdjust].slice(1)
+      );
+      const adjustedAxisIdNumber = currentAxisIdNumber - 1;
+
+      const currentAxisLayoutProperties = {
+        ...graphDiv.layout[payload.axisAttrToAdjust + currentAxisIdNumber],
+      };
+
+      // for cases when we're adjusting x2 => x, so that it becomes x not x1
+      graphDiv.data[trace.index][payload.axisAttrToAdjust] =
+        adjustedAxisIdNumber === 1 ? axis : axis + adjustedAxisIdNumber;
+
+      graphDiv.layout[
+        payload.axisAttrToAdjust + adjustedAxisIdNumber
+      ] = currentAxisLayoutProperties;
+    });
+  }
+};
+
+export const shamefullyAdjustGeo = ({layout: {geo = {}}}, {update}) => {
```

Comment:
@VeraZab this is finally how I got the geo stuff sorted. I also had to deal with the projection centers etc. Note that setting the value to `{}` essentially clears it, so this restores the defaults instead of hardcoding mappings, which is nice :)